clientv2/client.go

```diff
@@ -108,32 +111,179 @@ func (er *ErrorResponse) Error() string {
 	return string(content)
 }
 
-// the response into the given object.
+type MultipartFile struct {
+	File  graphql.Upload
+	Index int
+}
+
+type MultipartFilesGroup struct {
+	Files      []MultipartFile
+	IsMultiple bool
+}
+
+type FormField struct {
+	Name  string
+	Value interface{}
+}
+
+type header struct {
+	key, value string
+}
+
+// Post support send multipart form with files https://gqlgen.com/reference/file-upload/ https://github.com/jaydenseric/graphql-multipart-request-spec
 func (c *Client) Post(ctx context.Context, operationName, query string, respData interface{}, vars map[string]interface{}, interceptors ...RequestInterceptor) error {
+	multipartFilesGroups, mapping, vars := parseMultipartFiles(vars)
+
 	r := &Request{
 		Query:         query,
 		Variables:     vars,
 		OperationName: operationName,
 	}
+
 	gqlInfo := NewGQLRequestInfo(r)
+	body := new(bytes.Buffer)
+
+	var headers []header
+
+	if len(multipartFilesGroups) > 0 {
+		contentType, err := prepareMultipartFormBody(
+			body,
+			[]FormField{
+				{
+					Name:  "operations",
+					Value: r,
+				},
+				{
+					Name:  "map",
+					Value: mapping,
+				},
+			},
+			multipartFilesGroups,
+		)
```

Comment:
Could this possibly mean that for multipartFilesGroups, it is the only request that is supported?

Reply:
Fix multiple queries in a request
![image](https://user-images.githubusercontent.com/30795797/208027256-9f794442-3d44-42a3-a0b9-8efb32218ae1.png)

Reply:
/example/files-info/main.go - added examples client.AllFilesInfo(ctx, uFile, uFiles) and client.AllFilesInfoWithListItems(
		ctx,
		uFile,
		uFiles,
		&gen.ListItemsInput{IDAnyOf: []int{1, 3, 5}},
	)

Reply:
Does it work correctly for mutations?

Reply:
![image](https://user-images.githubusercontent.com/30795797/208059437-48e0dafd-24f2-43a2-aa77-52ed7682c838.png)

/example/files-info/main.go - added example `client.ManyMutationsInOne(ctx, uFile, uFiles, "New Item Name")`